extensions/amp-consent/0.1/consent-state-manager.js

```diff
@@ -204,6 +214,24 @@ export class ConsentStateManager {
     this.instance_.sharedDataPromise = sharedDataPromise;
   }
 
+  /**
+   * Signifies that we collected all the purpose consents
+   * from our source of truth (i.e locally stored or updated)
+   */
+  hasAllPurposeConsents() {
+    this.hasAllPurposeConsentsResolver_();
+  }
+
+  /**
+   * Wait for our purpose consents to be collected,
+   * either locally or from update. Let amp-consent handle
+   * when to resolve the promis (via update() or directly).
+   * @return {!Promise}
+   */
+  whenHasAllPurposeConsents() {
```

Comment:
nit: rename this method? Perhaps just `getHasAllPurposeConsents`?

Reply:
I agree the naming here can add some confusion (esp. since `hasFooBar` is frequently used for boolean flags/checks rather than mutation functions). WDYT about something like:
```javascript
// in constructor
this.hasAllPurposeConsentsDeferred = new Deferred();

get hasAllPurposeConsents() {
  return this.hasAllPurposeConsentsDeferred.promise;
}
get hasAllPurposeConsentsResolver() {
  return this.hasAllPurposeConsentsDeferred.resolve;
} 
```

A bit more verbose, but maybe being explicit could make it clearer?